src/compiler/transformers/ts.ts

```diff
@@ -2669,18 +2671,19 @@ export function transformTypeScript(context: TransformationContext) {
     function substituteConstantValue(node: PropertyAccessExpression | ElementAccessExpression): LeftHandSideExpression {
         const constantValue = tryGetConstEnumValue(node);
         if (constantValue !== undefined) {
-            // track the constant value on the node for the printer in needsDotDotForPropertyAccess
-            setConstantValue(node, constantValue);
+            const substitute = typeof constantValue === "string" ? factory.createStringLiteral(constantValue) :

```

Comment:
I think it's somewhat misleading that `resolver.shouldParenthesizeConstantValue(node)` gets only called for numbers here. The name of this function doesn't really imply that and I would expect it to be called for just any `constantValue` - then based on the result some conditional wrapping etc could happen here.